test/EzMap.IntegrationTest/PoiCollectionControllerTest.cs

```diff
@@ -76,6 +110,51 @@ public async Task UpdatePoiCollection_CorrectDataProvided_PoiColShouldBeUpdated(
         Assert.NotNull(dbPoiCol);
     }
 
+    [Fact]
+    public async Task UpdatePoiCollectio_EmptyPoiColIdProvided_EmptyIdErrorMessageShouldBeShowed()
+    {
+        var app = new TestWebAppFactory<Program>();
+        var client = app.CreateClient();
+        using var scope = app.Services.CreateScope();
+        var dbContext = scope.ServiceProvider.GetRequiredService<EzMapContext>();
+
+        var token = await TestHelper.GetDefaultUserToken(client);
+
+        var user = new User("thanh", "thanh", "thanh", "thanh");
+        dbContext.Users.Add(user);
+
+        var poiCol = new PoiCollection("home", "59 ntt", TestUser.DefaultUser.Id);
+        dbContext.PoiCollections.Add(poiCol);
+
+        await dbContext.SaveChangesAsync();
+
+        var updateDto = new PoiCollectionUpdateDto(
+            Guid.Empty,
+            "THANH NUMBER FAV PLACE",
+            "citygarden",
+            PoiEnum.ViewType.Map
+        );
+
+        var response =
+            await client.RequestAsJsonAsyncWithToken(HttpMethod.Put, $"api/poicollection/{poiCol.Id}", token,
+                updateDto);
+
+        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+        var comparingProperty = nameof(PoiCollectionUpdateDto.Id);
+
+        var responseString = await response.Content.ReadAsStringAsync();
+
+        var errorMessages = JsonSerializer.Deserialize<ProblemDetails>(responseString, new JsonSerializerOptions()
+        {
+            PropertyNameCaseInsensitive = true,
+        });
+
+        var idErrorMessage = errorMessages?.Errors[comparingProperty][0];
+
+        Assert.Contains($"{comparingProperty}".ToLower(), idErrorMessage?.ToLower());
```

Comment:
Same thing, there is error on Id, but what error, "Too long", "Too short"?

Anyways, you do not need to slap "ToLower" on every string comparison, that shows 2 things, 1stly, you don't even know what your program outputs/inputs. 2ndly, It's not what it's intended to be, for string comparison, use "str1".Equals("str2", StringComparison options). You can specify caseSensitivity in options as well as culture options.

Again, why does this tiny PR needs to be reviewed 2 times because of your laziness, think about it for a sec.

Reply:
Yes sir, will be more thoughtful next time!